### PR TITLE
Restructure behaviours

### PR DESCRIPTION
**Pull request by kanes115**
This PR proposes some changes in elements` behaviours names and relations between them. The changes are:
* Allow user to `use` elements with shorter module names. 
  * `Membrane.Element.Base.Source` -> `Membrane.Source`
  * `Membrane.Element.Base.Sink` -> `Membrane.Sink`
  * `Membrane.Element.Base.Filter` -> `Membrane.Filter`
* Change undescriptive behaviour names. 
  * `Membrane.Element.Base.Mixin.CommonBehaviour` -> `Membrane.Element.Base`
  * `Membrane.Element.Base.Mixin.SourceBehaviour` -> `Membrane.Element.WithOutputPads`
  * `Membrane.Element.Base.Mixin.SinkBehaviour` -> `Membrane.Element.WithInputPads`

### CODE REVIEW

File: lib/membrane/element/base.ex [hunk 1]
```diff
@@ -1,43 +1,228 @@
...
-  executing element-specific code.
-
-  For actions that can be returned by each callback, see `Membrane.Element.Action`
-  module.
+  Module defining behaviour common to all elements.
+
+  When used declares behaviour implementation, provides default callback definitions
+  and imports macros.
+
+  For more information on implementing elements, see `Membrane.Element.Base`.
```
Comment (mat-hek) [inline lib/membrane/element/base.ex]: This is `Membrane.Element.Base`...

File: lib/membrane/element/base.ex [hunk 2]
```diff
@@ -1,43 +1,228 @@
 defmodule Membrane.Element.Base do
   @moduledoc """
-  Modules in this namespace contain behaviours, default callback implementations
```
Comment (mat-hek) [inline lib/membrane/element/base.ex]: Not sure if this entire doc should be removed from here

**Comment by mat-hek** [comment]
Make `Membrane.Element.Base.*` delegate to `Membrane.*` and deprecate them

**Comment by kanes115** [comment]
Sorry, I requested review too early. Now it is ready to be reviewed, @mat-hek

**Pull request merged**